### PR TITLE
fix: "Open in VS Code" button for Controller Extension dialog does not work in BAS

### PR DESCRIPTION
**Pull request by nikmace**
Fix for #1750.

- Enhance route to return environment in the response to then dynamically show/hide button in the Controller dialog.

### CODE REVIEW

File: .changeset/six-eggs-eat.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+---
+'@sap-ux-private/preview-middleware-client': patch
+'@sap-ux/adp-tooling': patch
+'@sap-ux/preview-middleware': patch
+---
```
Comment (GDamyanov) [inline .changeset/six-eggs-eat.md:5]: Please add patch to "preview-middleware" as well

**Review by GDamyanov — CHANGES_REQUESTED** [review]

File: .changeset/six-eggs-eat.md (hunk 1 above)
Comment (nikmace) [reply .changeset/six-eggs-eat.md:5]: But it is there? :)

**Review by GDamyanov — APPROVED** [review]
Simple change ease to understand.
Good test coverage.
Manual testing done.

**Review by voicis — APPROVED** [review]
- Changeset ok
- Code adheres to coding standards
- Good test coverage
- Tested locally with mocking BAS environment

**Review by GDamyanov — APPROVED** [review]
Code looks good.
Manual testing done.

**Pull request merged**